### PR TITLE
Chore: No Repos Detected status message

### PR DESCRIPTION
**Pull request by st0nebraker**
Cleaner solution follow up to #[49348](https://github.com/sourcegraph/sourcegraph/pull/49348/files/08ddbf9fbd4712380ac1ef34aff17865d9e88838#r1142203764). Adds a new status message for `no repository present` state when user has no repositories and no sync jobs are happening.

## Test plan
- Test case added, ensure test passes
- QA: Delete code hosts and ensure status message updates to appropriate message:
image

### CODE REVIEW

**Comment by st0nebraker** [comment]
I will fix the bazel build error tomorrow because my internet is out with a wind storm right now, so git installs aren't working well on hotspot 😬

File: client/web/src/nav/StatusMessagesNavItem.tsx [hunk 1]
```diff
@@ -279,6 +266,19 @@ export const StatusMessagesNavItem: React.FunctionComponent<React.PropsWithChild
                             />
                         )
                     }
+                    if (status.__typename === 'NoRepositoriesDetected') {
+                        return (
+                            <StatusMessagesNavItemEntry
+                                key="no-repositories"
+                                title="No repositories detected"
```
Comment (mrnugget) [inline client/web/src/nav/StatusMessagesNavItem.tsx:273]:
```suggestion
                                title="No repositories"
```

I think that sounds more... human? 😄 We also say "no files in folder" and not "no files in folder detected", I thik.

**Review by mrnugget — APPROVED** [review]
VERY COOL! Is this your first Go PR? If so: congrats! :)

**Review by gl-srgr — APPROVED** [review]
LGTM

**Comment by st0nebraker** [comment]
> VERY COOL! Is this your first Go PR? If so: congrats! :)

It is my go at Go 😬  thanks!

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (+0.12 kb) | 0.00% (+0.23 kb) | 0.00% (+0.11 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/becca/new-status-msg/compare-report.html;tab=live_object) for a full comparison between the commits 2639410 and 5fca5ae or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Pull request merged**